### DISCUSSION TITLE
how to write a clean-up for withRepeat?

### DISCUSSION THREAD

**Discussion by malikKartik** [software-mansion/react-native-reanimated#2415] [Q&A]
Do I need to clean up withRepeat when a component is unmounted or before it will be rerendered?

I am using withRepeat for an ongoing animation, whenever my application refreshes(while development) a new instance of withRepeat is started on the same shared value which results in an unexpected animation. So, I can write a cleanup for this, my question is how can I write a clean-up for this or is it actually required to write a cleanup in this case as this will happen only in development mode?

Thanks

**Comment by Lewitje** [comment]
Better late than never!

`useEffect(() => {
    // Start your animation
    ....

    return () => {
        // Cancel your animation
        cancelAnimation(yourSharedValue)
        // Reset your shared value (helps with transforms)
        yourSharedValue.value = 0
    }
})`
